pkg/api/api.go

```diff
@@ -5,13 +5,13 @@ import (
 	"context"
 
 	acsapi "github.com/Azure/acs-engine/pkg/api"
+	"github.com/Azure/acs-engine/pkg/api/osa/vlabs"
 )
 
-type NewPlugin func(cs, oldCs *acsapi.ContainerService, configBytes []byte) (Plugin, error)
-
 type Plugin interface {
-	Validate() error
-	GenerateConfig() ([]byte, error)
-	GenerateARM() ([]byte, error)
-	HealthCheck(context.Context) error
+	ValidateExternal(oc *vlabs.OpenShiftCluster) []error
```

Comment:
@pweil- @amanohar changes to Plugin interface, are you OK with this?

Reply:
yeah, now that we've settled on shape and location I'm fine with this.  

it looks like this has also changed the shape to a stateless plugin.  Looks good